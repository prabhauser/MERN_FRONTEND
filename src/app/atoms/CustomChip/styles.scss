@import '../../../utilities/theme/_theme.scss';
@import '../../../assets/fonts/typography.scss';

.mainBadge {
    display: flex;
    width: 100%;
    @extend .c-poppins-semibold;
    font-size: 13px;
}
.badge {
    color: $white;
    background-color: $p800Bg;
    width: auto;
    @extend .c-poppins-semibold;
    // padding: 11px 20px 11px 20px;
    padding: 10px;
    border-radius: 25px;
    cursor:pointer;
    font-size: 13px;
    margin-right: 3%;
    min-width: 100px;
    display: flex;
    justify-content: center;
    margin-top: 10px;
}
.insideBadge {
    background-color: #03cbe9;
    color: $white;
    margin-left: 10px;
    border-radius: 20px;
    font-size: 11px;
    padding: 2px 10px;
}
.whiteBadge {
    background-color: $white;
    color: #bfc4cb;
    border: 1px solid lightgrey;
}

@media (max-width: 480px) {
    .mainBadge {
        display: flex;
        justify-content: center;
    }
    .badge {
        padding: 10px 17px 11px 17px;
        border-radius: 25px;
        font-size: 12px;
        margin-right: 3%;
        box-shadow: 0px 1px 2px 0px grey;
        display: flex;
        margin-top: 10px;
    }
}
@media (max-width: 767px) {
    .mainBadge {
        flex-wrap: wrap;
    }
}
.table_actionmenu:hover {
    background: transparent;
    color: $p800Bg;
}
