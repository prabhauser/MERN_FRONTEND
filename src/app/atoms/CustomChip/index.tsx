import theme from '../../../utilities/theme/theme.module.scss';
import './styles.scss';
import { useEffect, useState } from 'react';

type CustomChipType = {
    all: boolean;
    active: boolean;
    inactive: boolean;
    values: Number[];
    allText?: any;
    activeText?: any;
    inactiveText?: any;
    dotRequired: boolean;
    click?: any;
    resetBadge?: boolean;
};

const RedDot = () => {
    return (
        <div
            style={{
                width: '10px',
                height: '10px',
                backgroundColor: `${theme.red}`,
                borderRadius: '50px',
                marginTop: '2px',
                marginRight: '7px'
            }}
        ></div>
    );
};

const GreenDot = () => {
    return (
        <div
            style={{
                width: '10px',
                height: '10px',
                backgroundColor: `${theme.green}`,
                borderRadius: '50px',
                marginTop: '2px',
                marginRight: '7px'
            }}
        ></div>
    );
};

const ColorChips = (props: CustomChipType) => {
    const {
        all,
        active,
        inactive,
        values,
        allText,
        activeText,
        inactiveText,
        dotRequired,
        click = () => {},
        resetBadge = false
    } = props;
    const [activeTile, setActiveTile] = useState<string>('All');

    useEffect(() => {
        if (resetBadge && all) {
            setActiveTile('All');
            // click('All');
        } else if (resetBadge && !all) {
            setActiveTile('Active');
        }
    });

    return (
        <div className="mainBadge">
            {all ? (
                <span
                    className={
                        activeTile == 'All' || activeTile == 'Health' || activeTile == '' ? 'badge' : 'badge whiteBadge'
                    }
                    onClick={() => {
                        setActiveTile('All');
                        click('All');
                    }}
                >
                    {allText ? allText : 'All'}
                    <span className="insideBadge">{values[0]}</span>
                </span>
            ) : null}

            {active ? (
                <span
                    className={activeTile == 'Active' || activeTile == 'Eduation' ? 'badge' : 'badge whiteBadge'}
                    onClick={() => {
                        setActiveTile('Active');
                        click('ACTIVE');
                    }}
                >
                    {dotRequired ? <GreenDot /> : null}
                    {activeText ? activeText : 'Active'}
                    <span className="insideBadge">{values[1]}</span>
                </span>
            ) : null}

            {inactive ? (
                <span
                    className={activeTile == 'Inactive' || activeTile == 'FDS' ? 'badge' : 'badge whiteBadge'}
                    onClick={() => {
                        setActiveTile('Inactive');
                        click('INACTIVE');
                    }}
                >
                    {dotRequired ? <RedDot /> : null}
                    {inactiveText ? inactiveText : 'Inactive'}
                    <span className="insideBadge">{values[2]}</span>
                </span>
            ) : null}
        </div>
    );
};
export default ColorChips;
