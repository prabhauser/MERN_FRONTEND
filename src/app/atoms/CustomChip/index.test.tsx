import { shallow, mount } from 'enzyme';
import CustomChip from './index';
import { render, screen } from '@testing-library/react';

describe('CustomChip Component Tests', () => {
    it('should take CustomChip snapshot', () => {
        const component = shallow(
            <CustomChip all={true} active={active} inactive={inactive} values={values} dotRequired={false} />
        );
        component.update();
        expect(component).toMatchSnapshot();
    });
    let all = true;
    let active = false;
    let inactive = false;
    const values = [10, 20, 30];
    it('Expects to have All chip in the DOM', () => {
        render(<CustomChip all={true} active={active} inactive={inactive} values={values} dotRequired={false} />);
        const element = screen.getByText(/All/i);
        expect(element).toBeInTheDocument();
    });
    all = false;
    active = true;
    inactive = false;
    it('Expects to have Active chip in the DOM', () => {
        render(<CustomChip all={true} active={active} inactive={inactive} values={values} dotRequired={false} />);
        const element = screen.getByText(/Active/i);
        expect(element).toBeInTheDocument();
    });
    all = false;
    active = false;
    inactive = true;
    it('Expects to have Inactive chip in the DOM', () => {
        render(<CustomChip all={true} active={active} inactive={inactive} values={values} dotRequired={false} />);
        const element = screen.getByText(/Inactive/i);
        expect(element).toBeInTheDocument();
    });
    all = true;
    active = false;
    inactive = true;
    it('Expects to have both All and Inactive chips in the DOM', () => {
        render(<CustomChip all={true} active={active} inactive={inactive} values={values} dotRequired={false} />);
        const allElement = screen.getByText(/All/i);
        const inactiveElement = screen.getByText(/Inactive/i);
        expect(allElement).toBeInTheDocument();
        expect(inactiveElement).toBeInTheDocument();
    });
});
