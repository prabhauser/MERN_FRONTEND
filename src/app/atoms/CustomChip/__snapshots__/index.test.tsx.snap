// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomChip Component Tests should take CustomChip snapshot 1`] = `
<div
  className="mainBadge"
>
  <span
    className="badge"
    onClick={[Function]}
  >
    All
    <span
      className="insideBadge"
    >
      10
    </span>
  </span>
  <span
    className="badge whiteBadge"
    onClick={[Function]}
  >
    Inactive
    <span
      className="insideBadge"
    >
      30
    </span>
  </span>
</div>
`;
